src/main/java/com/yelp/nrtsearch/server/luceneserver/SearchHandler.java

```diff
@@ -965,6 +977,17 @@ else if (fd instanceof IndexableFieldDef && ((IndexableFieldDef) fd).isStored())
     return sortedFields;
   }
 
+  private int getNumHitsToCollect(int topHits, SearchRequest request) {
```

Comment:
Would be good to add some comments on why we are retrieving more than topHits for sample docs.